src/store/reducers/tableSchemaData.ts

```diff
@@ -0,0 +1,67 @@
+import {
+    prepareSchemaData,
+    prepareViewSchema,
+} from '../../containers/Tenant/Schema/SchemaViewer/prepareData';
+import type {SchemaData} from '../../containers/Tenant/Schema/SchemaViewer/types';
+import {isViewType} from '../../containers/Tenant/utils/schema';
+import type {EPathType} from '../../types/api/schema';
+import {isQueryErrorResponse} from '../../utils/query';
+
+import {api} from './api';
+import {overviewApi} from './overview/overview';
+import {viewSchemaApi} from './viewSchema/viewSchema';
+
+export interface GetTableSchemaDataParams {
+    path: string;
+    tenantName: string;
+    type: EPathType;
+}
+
+const TABLE_SCHEMA_TIMEOUT = 1000;
+
+const getTableSchemaDataConcurrentId = 'getTableSchemaData';
+
+export const tableSchemeDataApi = api.injectEndpoints({
+    endpoints: (build) => ({
+        getTableSchemaData: build.mutation<SchemaData[], GetTableSchemaDataParams>({
+            queryFn: async ({path, tenantName, type}, {dispatch}) => {
+                try {
+                    const schemaData = await dispatch(
```

Comment:
Do you really need this request for view? Probably, you should check if it's a table here

```js
if (isViewType(type)){
  // do request for a view
} else {
  // do request for a table
}
```

Reply:
fixed